Fix map preview not extended to full view

kosmydel opened the pull request:
<!-- If necessary, assign reviewers that know the area or changes well. Feel free to tag any additional reviewers you see fit. -->

### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please postfix `$` with a URL link to the GitHub issue this Pull Request is fixing. For example, `$ https://github.com/Expensify/App/issues/<issueID>`.
2. Please postfix  `PROPOSAL:` with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).  For example, `PROPOSAL: https://github.com/Expensify/App/issues/<issueID>#issuecomment-1369752925`

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking and its automation will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<issueID>
$ https://github.com/Expensify/App/issues/<issueID(comment)>

Do NOT only link the issue number like this: $ #<issueID>
--->
$ https://github.com/Expensify/App/issues/28911
$ https://github.com/Expensify/App/pull/26307#discussion_r1348040334
PROPOSAL: N/A


### Tests
<!---
Add a numbered list of manual tests you performed that validates your changes work on all platforms, and that there are no regressions present.
Add any additional test steps if test steps are unique to a particular platform.
Manual test steps should be written so that your reviewer can repeat and verify one or more expected outcomes in the development environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console
- [ ] Click Request Money
- [ ] Go to the distance tab
- [ ] Check the map appearance (verify border-radius, paddings, border-width, etc.)
- [ ] Select two random stops
- [ ] Press Next, select a workspace
- [ ] Check the map appearance (verify as above)
- [ ] **Do the same in the offline mode (so the Pending map view is displayed).**

### Offline tests
<!---
Add any relevant steps that validate your changes work as expected in a variety of network states e.g. "offline", "spotty connection", "slow internet", etc. Manual test steps should be written so that your reviewer and QA testers can repeat and verify one or more expected outcomes. If you are unsure how the behavior should work ask for advice in the `#expensify-open-source` Slack channel.
--->

### QA Steps
<!---
Add a numbered list of manual tests that can be performed by our QA engineers on the staging environment to validate that your changes work on all platforms, and that there are no regressions present.
Add any additional QA steps if test steps are unique to a particular platform.
Manual test steps should be written so that the QA engineer can repeat and verify one or more expected outcomes in the staging environment.

For example:
1. Click on the text input to bring it into focus
2. Upload an image via copy paste
3. Verify a modal appears displaying a preview of that image
--->

- [ ] Verify that no errors appear in the JS console

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

![Screenshot 2023-10-10 at 11 47 28](https://github.com/Expensify/App/assets/104823336/0181732a-9766-48d9-b479-b2bb136a49d9)
![Screenshot 2023-10-10 at 11 48 26](https://github.com/Expensify/App/assets/104823336/7ed813a4-79a4-4d88-97ae-4997fcdb9193)
![Screenshot 2023-10-10 at 11 48 16](https://github.com/Expensify/App/assets/104823336/31a61d90-0519-40e1-a910-b053973ccd17)
![Screenshot 2023-10-10 at 11 47 57](https://github.com/Expensify/App/assets/104823336/545619cf-b46d-4b55-b843-fad6c0e22f97)


**OLD:**

![Screenshot 2023-10-05 at 16 32 42](https://github.com/Expensify/App/assets/104823336/eba96a1c-f005-4a2a-a880-16e61deb93f7)
![Screenshot 2023-10-05 at 16 33 03](https://github.com/Expensify/App/assets/104823336/5b394ccc-9694-477c-82ff-368d76f7530c)

<img width="380" alt="Screenshot 2023-10-05 at 23 57 46" src="https://github.com/Expensify/App/assets/104823336/b5f0c155-e6f9-4efb-b92c-fc968cce9014">


</details>

<details>
<summary>Mobile Web - Chrome</summary>


**OLD:**

<!-- add screenshots or videos here -->
![Screenshot 2023-10-05 at 16 43 54](https://github.com/Expensify/App/assets/104823336/ce0c1b59-51f7-4f60-82c8-75199ed511fa)

![Screenshot 2023-10-05 at 16 44 04](https://github.com/Expensify/App/assets/104823336/053c34ad-8dfd-44b1-9e7b-0883d62ff524)

</details>

<details>
![Screenshot 2023-10-10 at 11 50 46](https://github.com/Expensify/App/assets/104823336/42f7335d-0a19-42f4-b032-165bb687c533)
![Screenshot 2023-10-10 at 11 51 16](https://github.com/Expensify/App/assets/104823336/1bfe9fa4-63d2-4ffb-aa05-ff1c6e726631)
![Screenshot 2023-10-10 at 11 51 20](https://github.com/Expensify/App/assets/104823336/e3b985cc-36e2-4dd7-8845-5e53ad4aeb23)

<summary>Mobile Web - Safari</summary>
![Screenshot 2023-10-10 at 11 50 41](https://github.com/Expensify/App/assets/104823336/3f7eb85b-455f-4d50-b3d3-11e4736fa1bd)


**OLD:**

<!-- add screenshots or videos here -->
![Screenshot 2023-10-05 at 16 44 12](https://github.com/Expensify/App/assets/104823336/cb4e4fa0-8c69-4ad6-b822-1ea02d64c3b8)


<img width="403" alt="Screenshot 2023-10-06 at 00 00 01" src="https://github.com/Expensify/App/assets/104823336/959a3a9d-93d2-4c4b-bdea-32f2f0fd454c">



</details>

<details>
<summary>Desktop</summary>


**OLD:**

<!-- add screenshots or videos here -->
![Screenshot 2023-10-05 at 16 37 46](https://github.com/Expensify/App/assets/104823336/38a71782-bd67-4969-8a80-c3b7d75b39ba)

![Screenshot 2023-10-05 at 16 37 43](https://github.com/Expensify/App/assets/104823336/8e949af7-b106-4756-9568-e332096f501d)

</details>

<details>
<summary>iOS</summary>

![Screenshot 2023-10-10 at 11 49 19](https://github.com/Expensify/App/assets/104823336/950409cd-5d22-4197-9b83-9ba0c4dec959)
![Screenshot 2023-10-10 at 11 49 28](https://github.com/Expensify/App/assets/104823336/d98d2946-6c2f-42eb-a139-0f96f6884523)
![Screenshot 2023-10-10 at 11 49 38](https://github.com/Expensify/App/assets/104823336/bbbf43dd-da89-468e-85f3-fc6e34980a2d)
![Screenshot 2023-10-10 at 11 49 43](https://github.com/Expensify/App/assets/104823336/e77ba82c-e495-4963-b9f1-64abe228ec92)


**OLD:**

<!-- add screenshots or videos h
ere -->

![Screenshot 2023-10-05 at 16 39 49](https://github.com/Expensify/App/assets/104823336/2a10dc02-b22e-4025-9cf5-5987e7e06809)

![Screenshot 2023-10-05 at 16 40 00](https://github.com/Expensify/App/assets/104823336/6e03f239-46fa-4c56-bd20-4d77ea6e9f9d)

<img width="408" alt="Screenshot 2023-10-05 at 23 58 39" src="https://github.com/Expensify/App/assets/104823336/214081df-f627-47ee-878a-68c2273c1370">


</details>

<details>
<summary>Android</summary>

![Screenshot 2023-10-10 at 11 53 23](https://github.com/Expensify/App/assets/104823336/ec91bd29-90d2-46f7-9d03-0f6764a2144f)
![Screenshot 2023-10-10 at 11 54 13](https://github.com/Expensify/App/assets/104823336/1d1c7431-3077-4855-b5a5-a45560290b14)
![Screenshot 2023-10-10 at 11 54 25](https://github.com/Expensify/App/assets/104823336/3c9f9aec-8829-41cd-b7fd-7ca3dfc65f91)
![Screenshot 2023-10-10 at 11 54 35](https://github.com/Expensify/App/assets/104823336/61161fdb-93d5-47e2-82c4-313c18a1410e)



**OLD:**

![Screenshot 2023-10-05 at 16 40 56](https://github.com/Expensify/App/assets/104823336/e01e7c50-a332-43a1-81c3-242c46669493)

<!-- add screenshots or videos here -->
![Screenshot 2023-10-05 at 16 40 36](https://github.com/Expensify/App/assets/104823336/8a62c8b1-5be0-4b50-89e2-42bc339930a3)

</details>

kosmydel committed:
fix

kosmydel committed:
fix native

situchan commented:
@jjcoffee I'll be reviewing this as this stemmed from the PR I reviewed

shawnborton commented:
That looks good to me. I thought we also decided not to disable the submit button for the form submission. Do we know if that is being handled anywhere? cc @JmillsExpensify @trjExpensify @tgolen

shawnborton commented:
@kosmydel can you also share what this looks like when we have say 5-6 waypoints?

situchan commented:
> That looks good to me. I thought we also decided not to disable the submit button for the form submission. Do we know if that is being handled anywhere? cc @JmillsExpensify @trjExpensify @tgolen

yes - https://github.com/Expensify/App/issues/27045

kosmydel commented:
> @kosmydel can you also share what this looks like when we have say 5-6 waypoints?

Sure @shawnborton, here are some screens (I will add them to the description as well):

<details>
<summary>Web</summary>

<img width="380" alt="Screenshot 2023-10-05 at 23 57 46" src="https://github.com/Expensify/App/assets/104823336/febf3429-af5d-4104-874f-41ba89aa4b6e">


</details>

<details>
<summary>iOS</summary>

<img width="408" alt="Screenshot 2023-10-05 at 23 58 39" src="https://github.com/Expensify/App/assets/104823336/86c4ffce-5e02-4cbc-a106-e96955ce8209">


</details>

<details>
<summary>mWeb - iOS</summary>

<img width="403" alt="Screenshot 2023-10-06 at 00 00 01" src="https://github.com/Expensify/App/assets/104823336/46008fb0-e8e5-4119-8e54-1bb2c3767d9e">


</details>

shawnborton commented:
Nice, those look good. Can you show me what it looks like when you scroll down, too?

kosmydel committed:
cleanup

kosmydel committed:
Merge branch 'expensify-main' into fix-scrollview-container

kosmydel committed:
Merge branch 'main' into fix-scrollview-container

kosmydel committed:
Merge branch 'main' into fix-scrollview-container

kosmydel committed:
fix scrolling on the native

kosmydel commented:
I've added the cleanup change mentioned [here](https://github.com/Expensify/App/pull/26307#discussion_r1348040334) to this PR.

--- 

> Nice, those look good. Can you show me what it looks like when you scroll down, too?

Sure @shawnborton!

<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/104823336/fb9341f8-5f92-49bf-b9fc-cdd3feb2fd96



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/104823336/7b918ca7-52b5-4aac-a806-687b0a7f4ace


</details>

<details>
<summary>mWeb - iOS</summary>


https://github.com/Expensify/App/assets/104823336/25560d81-d3ca-4763-bece-887cfd5b6f4c



</details>

<details>
<summary>mWeb - Android</summary>




https://github.com/Expensify/App/assets/104823336/5a2a4346-04a9-4a87-a10d-2f0509c881e5




</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/104823336/37fafa73-8001-4c23-ac1f-94079c74ac72



</details>

kosmydel commented on src/components/DistanceRequest/index.js:
```diff
@@ -229,7 +229,4 @@ export default withOnyx({
     transaction: {
         key: ({transactionID}) => `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID || 0}`,
     },
-    mapboxAccessToken: {
-        key: ONYXKEYS.MAPBOX_ACCESS_TOKEN,
-    },
```
This is a cleanup change, discussed [here](https://github.com/Expensify/App/pull/26307#discussion_r1347867780).

tgolen replied:
Thank you!

shawnborton commented:
Nice, that looks good to me! The only minor thing here is that at some point, the map area became skinnier than we wanted. cc @neil-marcellini as I know you have noticed this too. Basically it should look like this, which should match the width we are using for the camera area of the Scan page:
<img width="1066" alt="CleanShot 2023-10-09 at 08 30 47@2x" src="https://github.com/Expensify/App/assets/2319350/dad47afc-15c3-4c7d-9ed9-6b71c99fb1f7">

kosmydel committed:
change padding

kosmydel commented:
@shawnborton  I changed `padding` from `16px` to `8px`.

Here is how it looks:

<details>
<summary>Web</summary>

![Screenshot 2023-10-09 at 15 02 05](https://github.com/Expensify/App/assets/104823336/118b3792-dd32-4f69-a6c3-b3d669c7fb06)


</details>

<details>
<summary>iOS</summary>

<img width="405" alt="Screenshot 2023-10-09 at 14 59 51" src="https://github.com/Expensify/App/assets/104823336/414bd41f-3cc0-4f04-92ec-84f1368b65fe">

</details>

<details>
<summary>mWeb - iOS</summary>

![Screenshot 2023-10-09 at 15 01 41](https://github.com/Expensify/App/assets/104823336/472d710b-620b-47d5-a0ff-56d7b980e3a6)


</details>

shawnborton commented:
Nice! Can you confirm that's exactly what we have on the Scan page, as well as the same border radius we're using there? Thanks!

kosmydel commented:
> Nice! Can you confirm that's exactly what we have on the Scan page, as well as the same border radius we're using there? Thanks!

I will check it tomorrow, as I'm finishing my work for today.

kosmydel committed:
match styles with scan view

kosmydel commented:
Hey @shawnborton, I've updated the PR. I've added new QA steps and new screenshots. I had to make some more changes, because of how the Scan page handled the radius. Now it should look exactly as on the Scan page.

<details>
<summary>Web</summary>

![Screenshot 2023-10-10 at 11 47 28](https://github.com/Expensify/App/assets/104823336/0181732a-9766-48d9-b479-b2bb136a49d9)
![Screenshot 2023-10-10 at 11 48 26](https://github.com/Expensify/App/assets/104823336/7ed813a4-79a4-4d88-97ae-4997fcdb9193)
![Screenshot 2023-10-10 at 11 48 16](https://github.com/Expensify/App/assets/104823336/31a61d90-0519-40e1-a910-b053973ccd17)
![Screenshot 2023-10-10 at 11 47 57](https://github.com/Expensify/App/assets/104823336/545619cf-b46d-4b55-b843-fad6c0e22f97)

</details>

<details>
<summary>Mobile Web - Safari</summary>

![Screenshot 2023-10-10 at 11 50 46](https://github.com/Expensify/App/assets/104823336/42f7335d-0a19-42f4-b032-165bb687c533)

![Screenshot 2023-10-10 at 11 51 16](https://github.com/Expensify/App/assets/104823336/1bfe9fa4-63d2-4ffb-aa05-ff1c6e726631)

![Screenshot 2023-10-10 at 11 51 20](https://github.com/Expensify/App/assets/104823336/e3b985cc-36e2-4dd7-8845-5e53ad4aeb23)

![Screenshot 2023-10-10 at 11 50 41](https://github.com/Expensify/App/assets/104823336/3f7eb85b-455f-4d50-b3d3-11e4736fa1bd)

</details>

<details>
<summary>iOS</summary>

![Screenshot 2023-10-10 at 11 49 19](https://github.com/Expensify/App/assets/104823336/950409cd-5d22-4197-9b83-9ba0c4dec959)
![Screenshot 2023-10-10 at 11 49 28](https://github.com/Expensify/App/assets/104823336/d98d2946-6c2f-42eb-a139-0f96f6884523)
![Screenshot 2023-10-10 at 11 49 38](https://github.com/Expensify/App/assets/104823336/bbbf43dd-da89-468e-85f3-fc6e34980a2d)
![Screenshot 2023-10-10 at 11 49 43](https://github.com/Expensify/App/assets/104823336/e77ba82c-e495-4963-b9f1-64abe228ec92)
</details>

<details>
<summary>Android</summary>

![Screenshot 2023-10-10 at 11 53 23](https://github.com/Expensify/App/assets/104823336/ec91bd29-90d2-46f7-9d03-0f6764a2144f)
![Screenshot 2023-10-10 at 11 54 13](https://github.com/Expensify/App/assets/104823336/1d1c7431-3077-4855-b5a5-a45560290b14)
![Screenshot 2023-10-10 at 11 54 25](https://github.com/Expensify/App/assets/104823336/3c9f9aec-8829-41cd-b7fd-7ca3dfc65f91)
![Screenshot 2023-10-10 at 11 54 35](https://github.com/Expensify/App/assets/104823336/61161fdb-93d5-47e2-82c4-313c18a1410e)

</details>

jjcoffee commented:
Can someone remove me as a reviewer and add @situchan instead, per [this comment](https://github.com/Expensify/App/pull/28922#issuecomment-1749151806)? Thanks!

neil-marcellini commented:
Tagging in Rocio for review too since she's managing the issue.

neil-marcellini commented:
Ah wait never mind that will happen after C+ approves.

situchan commented:
reviewing this tomorrow

kosmydel committed:
Merge branch 'main' into fix-scrollview-container

kosmydel committed:
refactor styling

kosmydel commented on src/styles/styles.js:
```diff
@@ -3680,16 +3680,25 @@ const styles = (theme) => ({
+    },
+
+    mapEditView: {
+        borderRadius: 28,
```
Sure! I've moved it

kosmydel commented on src/styles/styles.js:
```diff
@@ -3680,16 +3680,25 @@ const styles = (theme) => ({
+        borderColor: theme.appBG,
+    },
+
+    mapConfirmationView: {
```
That's right! I've removed other styles while developing, and didn't catch that!

kosmydel commented:
I've refactored it. Here is a video presenting both the map and its pending view:


<details>
<summary>
Video
</summary>

https://github.com/Expensify/App/assets/104823336/4bf58f16-0faf-4eb6-aac0-7e381d352f43

</details>

tgolen reviewed (dismissed):
Great, thanks for refactoring those styles a little.

tgolen commented:
There are some conflicts on here now. @situchan this is waiting on your final review.

situchan commented:
I will checklist once conflict resolved

kosmydel committed:
Merge branch 'main' into fix-scrollview-container

kosmydel commented:
Conflicts resolved, and quickly tested it

cc @situchan

kosmydel committed:
fix merge

kosmydel committed:
fix merge

pecanoro commented:
@situchan can you do a review as soon as possible? It's a high priority issue and it has been waiting for 3 days.

situchan commented:
reviewing now

situchan commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>


https://github.com/Expensify/App/assets/108292595/e1dc0606-4f3a-4e7f-a6fc-d6048402e203



</details>

<details>
<summary>Mobile Web - Chrome</summary>


https://github.com/Expensify/App/assets/108292595/027aa315-1f6b-4ce7-955f-ba73d69ddc1f



</details>

<details>
<summary>Mobile Web - Safari</summary>


https://github.com/Expensify/App/assets/108292595/20f35558-d948-417c-ba18-2b55023f46c8



</details>

<details>
<summary>Desktop</summary>


https://github.com/Expensify/App/assets/108292595/1cf5ff75-fc5d-4636-a866-daef1eb618e3



</details>

<details>
<summary>iOS</summary>


https://github.com/Expensify/App/assets/108292595/09251db4-e3df-4bba-84d9-ad6f2565fd88



</details>

<details>
<summary>Android</summary>


https://github.com/Expensify/App/assets/108292595/5257bb45-e30f-47a6-aecc-2314962e6f05



</details>

situchan reviewed (approved):
Pulled main into this branch locally and tests well (some known bugs are gone).

blazejkustra reviewed (approved):
LGTM, code looks good to me!

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6592629328) to staging by https://github.com/tgolen in version: 1.3.88-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/6608779761) to production by https://github.com/chiragsalian in version: 1.3.88-11 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|failure ❌
🕸 web 🕸|success ✅
